### PR TITLE
Fix TOC for /manual and /style_guide

### LINKED ISSUE #24: Table of Contents in /manual and /style_guide is broken

### PR DESCRIPTION
**Pull request by dshubhadeep**
I have used [markdown-toc](https://github.com/jonschlinkert/markdown-toc) to generate the table of contents for both the markdown files.

I have also added a custom heading renderer since `react-markdown` doesn't add anchor tags to headings by default. [See here](https://github.com/rexxars/react-markdown/issues/69)

Closes #24

### CODE REVIEW

**Comment by ry** [comment]
Thanks - But now there are two markdown parsers..?

File: src/manual.md [hunk 1]
```diff
@@ -1,6 +1,58 @@
...
+  - [Deno Core](#deno-core)
+  - [Updating prebuilt binaries](#updating-prebuilt-binaries)
+  - [Continuous Benchmarks](#continuous-benchmarks)
+  - [Logos](#logos)
+- [Contributing](#contributing)
+  - [Submitting a pull request](#submitting-a-pull-request)
+  - [Changes to `third_party`](#changes-to-third_party)
+  - [Adding Ops (aka bindings)](#adding-ops-aka-bindings)
+  - [Documenting APIs](#documenting-apis)
+    - [Utilize JSDoc](#utilize-jsdoc)
```
Comment (ry) [inline src/manual.md]: I’d rather this done at runtime

Comment (dshubhadeep) [reply src/manual.md]: Fixed it. I have added the [remark-toc](https://github.com/remarkjs/remark-toc) plugin.

It'll generate the TOC at runtime

File: src/Markdown.js [hunk 2]
```diff
@@ -1,11 +1,44 @@
...
 
-const renderers = { code: CodeBlock };
+function flatten(text, child) {
+  return typeof child === "string"
+    ? text + child
+    : React.Children.toArray(child.props.children).reduce(flatten, text);
+}
+
+// Using logic from markdown-toc
+const slugify = text => {
```
Comment (ry) [inline src/Markdown.js]: Please follow the Deno style guide ("Top level functions should not use arrow syntax
") and use `function slugify(text) {` here

File: src/Markdown.js [hunk 3]
```diff
@@ -1,11 +1,44 @@
...
 import CodeBlock from "./CodeBlock";
 
-const renderers = { code: CodeBlock };
+function flatten(text, child) {
+  return typeof child === "string"
+    ? text + child
+    : React.Children.toArray(child.props.children).reduce(flatten, text);
+}
+
+// Using logic from markdown-toc
```
Comment (ry) [inline src/Markdown.js]: Not necessary

File: src/Markdown.js [hunk 4]
```diff
@@ -1,11 +1,44 @@
...
+    .split(/[。？！，、；：“”【】（）〔〕［］﹃﹄“ ”‘’﹁﹂—…－～《》〈〉「」]/)
+    .join("");
+
+  return text;
+};
+
+function HeadingRenderer(props) {
+  var children = React.Children.toArray(props.children);
+  var text = children.reduce(flatten, "");
+  var slug = slugify(text);
```
Comment (ry) [inline src/Markdown.js]: s/var/const/

Comment (ry) [reply src/Markdown.js]: s/slug/id/

**Pull request merged**